Source/WebKit/UIProcess/Extensions/Cocoa/WebExtensionContextCocoa.mm

```diff
@@ -2951,18 +2952,17 @@ static _WKWebExtensionContextError toAPI(WebExtensionContext::Error error)
 
     m_declarativeNetRequestDynamicRulesStore = nullptr;
     m_declarativeNetRequestSessionRulesStore = nullptr;
-    m_declarativeNetRequestRuleStore = nullptr;
 }
 
-WKContentRuleListStore *WebExtensionContext::declarativeNetRequestRuleStore()
+String WebExtensionContext::declarativeNetRequestContentRuleListFilePath()
 {
-    if (m_declarativeNetRequestRuleStore)
-        return m_declarativeNetRequestRuleStore.get();
+    if (!m_declarativeNetRequestContentRuleListFilePath.isEmpty())
+        return m_declarativeNetRequestContentRuleListFilePath;
 
-    auto contentBlockerStorePath = storageIsPersistent() ? storageDirectory() : String(FileSystem::createTemporaryDirectory(@"DeclarativeNetRequest"));
-    m_declarativeNetRequestRuleStore = [WKContentRuleListStore storeWithURL:[NSURL fileURLWithPath:contentBlockerStorePath]];
+    auto directoryPath = storageIsPersistent() ? storageDirectory() : String(FileSystem::createTemporaryDirectory(@"DeclarativeNetRequest"));
+    m_declarativeNetRequestContentRuleListFilePath = FileSystem::pathByAppendingComponent(directoryPath, "DeclarativeNetRequestContentRuleList.data"_s);
```

Comment:
Is there existing data on people's disks that will be lost because of this file storage change?

Reply:
Only the 3-4 of us working on the extensions code.